sporkfish/transposition_table.py

```diff
@@ -1,40 +1,49 @@
 from typing import Dict, Optional
 
+import numpy as np
+
 
 class TranspositionTable:
-    def __init__(self, dct: Dict[int, Dict[str, float]]) -> None:
+    def __init__(self, dct: Dict[np.int64, Dict[str, float]]) -> None:
         """
-        Initialize the TranspositionTable object, wrapping a shared_dict.
+        Initialize the TranspositionTable object.
 
+        :param dct: A dictionary containing Zobrist hash keys and associated entries.
+        :type dct: Dict[np.int64, Dict[str, float]]
         """
         self._table = dct
 
     def store(
         self,
-        zobrist_hash: int,
+        zobrist_hash: np.int64,
         depth: int,
         score: float,
     ) -> None:
         """
         Store an entry in the transposition table.
-
-        Parameters:
-        - zobrist_hash (int): The Zobrist hash value for the board position.
-        - depth (int): The depth at which the score was calculated.
-        - score (float): The score associated with the board position.
+        Only stores if the existing entry depth is lower than the input one.
```

Comment:
Nice change - ty for cleaning up inconsistent docstring styles

Reply:
This comment seems to contradict with the code subtlety?

The code:
```
if not existing_entry or depth > existing_entry["depth"]:
```
My interpretation of the comment `Only stores if the existing entry depth is lower than the input one`:
```
if not existing_entry and depth > existing_entry["depth"]:
```

I suspect and `and` clause is what you wanted? As you only want to overwrite existing entries when we are visiting higher level (nearer to the top) node? 

I might be understanding this wrongly - lmk

Reply:
The original is correct no?

If not there, overwrite
If there, overwrite if depth > stored depth

Reply:
ahh yes ofc, I was tilting